spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/client/OAuth2ClientPropertiesRegistrationAdapter.java

```diff
@@ -75,7 +75,8 @@ private static ClientRegistration getClientRegistration(String registrationId,
 
 	private static Builder getBuilderFromIssuerIfPossible(String registrationId, String configuredProviderId,
 			Map<String, Provider> providers) {
-		String providerId = (configuredProviderId != null) ? configuredProviderId : registrationId;
+		String providerId = StringUtils
+				.trimWhitespace((configuredProviderId != null) ? configuredProviderId : registrationId);
```

Comment:
I think this logic can be moved a level up to avoid duplication of the trimming logic?

Reply:
Do we need to trim `registrationId` as well?

Reply:
updated

Reply:
I suppose it doesn't hurt to trim it but I don't think it's possible to have whitespace at the end of the `registrationId` since it's a key in the map.

Reply:
<  I don't think it's possible to have whitespace at the end of the registrationId since it's a key in the map

I've just realized and removed my last comment.